src/frontends/onnx/frontend/src/op/cum_sum.cpp

```diff
@@ -20,7 +20,12 @@ OutputVector cum_sum(const Node& node) {
     Output<ngraph::Node> axis;
 
     if (inputs.size() > 1) {
-        axis = inputs.at(1);  // optional input, 0-D tensor
+        // optional input, 0-D or 1-D tensor
+        const auto& axis_shape = inputs.at(1).get_partial_shape();
+        NGRAPH_CHECK(axis_shape.rank().get_length() == 0 || axis_shape.rank().get_length() == 1,
+                     "axis shape should be 0-D or 1-D.");
+        axis =
+            axis_shape.rank().get_length() == 0 ? inputs.at(1) : std::make_shared<default_opset::Squeeze>(inputs.at(1));
```

Comment:
To avoid issues `get_legth()` shouldn't be called without dynamism check, 
my suggestion is to check the shape and call the helper from `#include "utils/reshape.hpp"`:

```suggestion
            axis_shape.is_dynamic() ? inputs.at(1) : reshape::interpret_as_scalar(inputs.at(1));
```